### PR TITLE
Change export from owncloud to box

### PR DESCRIPTION
**Pull request by Dantemss**
Needs secrets added to tutor-deployment

### CODE REVIEW

File: app/controllers/admin/research_data_controller.rb [hunk 1]
```diff
@@ -20,15 +20,17 @@ def create
...
+          mapping.fetch(task_types_param) { |key| raise "Don't know about application #{key}" }
+        )
+      end
 
       from_date = params[:export_research_data][:from] || "1/1/1970"
       to_date = params[:export_research_data][:to] || Time.current.to_s
       ExportAndUploadResearchData.perform_later(filename: filename, from: from_date, to: to_date, task_types: task_types)
       redirect_to admin_research_data_path,
-                  notice: "#{ExportAndUploadResearchData::RESEARCH_FOLDER}/#{filename} should be available in a few minutes in ownCloud (does not refresh automatically)"
+                  notice: "#{filename} should be available in a few minutes in the exports folder"
```
Comment (jpslav) [inline app/controllers/admin/research_data_controller.rb]: may not be a few minutes :-)

File: app/controllers/manager/stats_actions.rb [hunk 2]
```diff
@@ -38,9 +38,9 @@ def excluded_exercises_to_csv
     end
 
     ExportExerciseExclusions.perform_later(
-      upload_by_course_to_owncloud: by_course, upload_by_exercise_to_owncloud: by_exercise
+      upload_by_course: by_course, upload_by_exercise: by_exercise
     )
-    flash[:success] = "The export should be available in a few minutes in ownCloud."
+    flash[:success] = "The export should be available in a few minutes in Box."
```
Comment (jpslav) [inline app/controllers/manager/stats_actions.rb]: ditto few minutes maybe not right - when it doesn't happen this fast researchers can rerun

File: spec/cassettes/Box/can_upload_files_to_box.yml [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+      string: ''
+    headers:
+      User-Agent:
+      - HTTPClient/1.0 (2.8.3, ruby 2.2.3 (2015-08-18))
+      Accept:
+      - "*/*"
+      Date:
+      - Wed, 27 Sep 2017 17:34:57 GMT
+      Authorization:
+      - Bearer kifioED9EHqAiR98fEeSAHVh8LMMhxi7
```
Comment (jpslav) [inline spec/cassettes/Box/can_upload_files_to_box.yml]: this one?

File: spec/cassettes/Box/can_upload_files_to_box.yml [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+      string: '{"name":"excluded_exercises_stats_by_course_20170925T213441Z.csv","parent":{"id":"39095941777"},"size":1003}'
+    headers:
+      User-Agent:
+      - HTTPClient/1.0 (2.8.3, ruby 2.2.3 (2015-08-18))
+      Accept:
+      - "*/*"
+      Date:
+      - Wed, 27 Sep 2017 17:34:57 GMT
+      Authorization:
+      - Bearer kifioED9EHqAiR98fEeSAHVh8LMMhxi7
```
Comment (jpslav) [inline spec/cassettes/Box/can_upload_files_to_box.yml]: ditto

File: spec/cassettes/Box/can_upload_files_to_box.yml [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      encoding: UTF-8
+      string: '{"upload_url":"https:\/\/upload.box.com\/api\/2.0\/files\/content?upload_session_id=3ED1AEBEA65DE62AD9F9F98737332F16","upload_token":null}'
+    http_version:
+  recorded_at: Wed, 27 Sep 2017 17:34:58 GMT
+- request:
+    method: post
+    uri: https://upload.box.com/api/2.0/files/content
+    body:
+      encoding: UTF-8
+      string: "--a33324b20b4b4f07cceed330da60db782cd9e451\r\nContent-Disposition:
```
Comment (jpslav) [inline spec/cassettes/Box/can_upload_files_to_box.yml]: what is a333...

File: spec/cassettes/Box/can_upload_files_to_box.yml [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+        null\"\n\r\n--a33324b20b4b4f07cceed330da60db782cd9e451--\r\n\r\n"
+    headers:
+      User-Agent:
+      - HTTPClient/1.0 (2.8.3, ruby 2.2.3 (2015-08-18))
+      Accept:
+      - "*/*"
+      Date:
+      - Wed, 27 Sep 2017 17:34:58 GMT
+      Authorization:
+      - Bearer kifioED9EHqAiR98fEeSAHVh8LMMhxi7
```
Comment (jpslav) [inline spec/cassettes/Box/can_upload_files_to_box.yml]: ditto

File: spec/cassettes/Box/can_upload_files_to_box.yml (hunk 5 above)
Comment (Dantemss) [reply spec/cassettes/Box/can_upload_files_to_box.yml]: No idea... comes from the gem

**Pull request merged**